internal/completions/tokenizer/tokenizer.go

```diff
@@ -69,49 +60,34 @@ func modelFamilyFromString(model string) ModelFamily {
 	switch {
 	case strings.Contains(model, AnthropicModel):
 		switch {
-		// Selects for claude 2, claude 2.1 and claude instant models
 		case strings.Contains(model, "claude-3"):
 			return Claude3
 		default:
-			// Selects for claude 3 models by exclusion
+			// Claude 2 models by exclusion
 			return Claude2
 		}
 	case strings.Contains(model, OpenAIModel), strings.Contains(model, AzureModel):
-		// Both Azure and OpenAI models use the same tokenizer
 		return GPT
 	default:
 		return UnknownModel
 	}
 }
 
-// NewTokenizer returns a Tokenizer instance based on the provided model.
+// NewTokenizer returns a cl100k_base Tokenizer instance for all models, use for abuse detection only.
+//
+// NOTE: The tokenizer must match the tokenizer used by the Cody clients to ensure consistency across clients,
+// and the Cody clients have standardized on using the cl100k_base tokenizer for token counting.
 func NewTokenizer(model string) (Tokenizer, error) {
-	modelFamily := modelFamilyFromString(model)
-	switch modelFamily {
-	case Claude2:
-		return newAnthropicClaudeTokenizer(model, Claude2)
-	case Claude3:
-		// Claude 3 models are not supported yet so tokenization will eventually need to be implemented
-		return newAnthropicClaudeTokenizer(model, Claude3)
-	case GPT:
-		return newOpenAITokenizer(model, GPT)
-	default:
-		return nil, errors.New("tokenizer not found for this model")
-	}
-}
-
-func newOpenAITokenizer(model string, modelFamily ModelFamily) (*tiktokenTokenizer, error) {
-	// Remove "azure" or "openai" prefix from the model string
+	// Remove "azure" or "openai" prefix from the model string if any
 	model = strings.NewReplacer(AzureModel+"/", "", OpenAIModel+"/", "").Replace(model)
+	modelFamily := modelFamilyFromString(model)
 
-	tkm, err := tiktoken.EncodingForModel(model)
+	// Use the offline loader to avoid downloading the encoding at runtime
```

Comment:
For readability, comments in code should be complete sentences. Meaning starting with a capital letter and ending with a period. (If the comment is only 2-3 words, don't worry about it.)

But the comment on line 81 needs a period, too.

```suggestion
	// Use the offline loader to avoid downloading the encoding at runtime.
```